raster-foundry/raster-foundry
# Project settings pages

## Overview

This PR adds the three child pages to the project settings page for the layers UI:
 - general project options (renaming and deleting)
 - publishing (tile layer urls, project visibility, etc)
 - permissions

### Checklist


### Demo

![image](https://user-images.githubusercontent.com/2442245/52803243-a62c9500-304f-11e9-928c-51e5d222674d.png)

![image](https://user-images.githubusercontent.com/2442245/52803293-c65c5400-304f-11e9-87f2-4c51f2b57dab.png)

![image](https://user-images.githubusercontent.com/2442245/52803323-d83df700-304f-11e9-94ab-440d176fcb18.png)

![image](https://user-images.githubusercontent.com/2442245/52803531-408cd880-3050-11e9-8a7b-aae975331b10.png)

![image](https://user-images.githubusercontent.com/2442245/52803647-7205a400-3050-11e9-8e14-2bbdc9582158.png)



### Notes

- This doesn't include the layer thumbnails or the tile URLs for a layer's analyses. I'm going to make issues for those as we've already spent more time on this than anticipated.
- You'll see usage of a factory and `ng-controller`. This was done so that we could re-use the logic within the permission UI controller. Having two almost identical controllers would have been asking for problems.
- The permissions UI works a little differently than the modal in that changes are immediately saved instead of having a save and cancel button

## Testing Instructions

 Test each of the pages. The desired functionality is not new, so it should be clear what things to try on these pages and how to verify that they work as expected.

Closes #4543
